pkg/apis/proxy/aws.go

```diff
@@ -0,0 +1,71 @@
+package proxy
+
+import (
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+	"github.com/gin-gonic/gin"
+
+	"github.com/seal-io/walrus/pkg/dao/model"
+	optypes "github.com/seal-io/walrus/pkg/operator/types"
+)
+
+type awsProxy struct {
+	conn    *model.Connector
+	context *gin.Context
+}
+
+func (p *awsProxy) handle(requestPath, rawQuery string) {
+	requestPath = appendSchemaIfMissing(requestPath)
+
+	u, err := url.Parse(requestPath)
+	if err != nil {
+		_ = p.context.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	if !isWhiteListDomain(u.Hostname()) {
+		p.context.AbortWithStatus(http.StatusForbidden)
+		return
+	}
+
+	cred, err := optypes.GetCredential(p.conn.ConfigData)
+	if err != nil {
+		_ = p.context.AbortWithError(http.StatusBadRequest, err)
+		return
+	}
+
+	rp := &httputil.ReverseProxy{
+		Director: func(request *http.Request) {
+			request.URL = u
+			request.Host = u.Host
+			request.URL.RawQuery = rawQuery
+
+			request.Header.Del("Connection")
```

Comment:
Need to pay more attention to the headers while proxy, don't forward the walrus cookie to the thirdparty, and remove the cookies that may affect walrus, like header set-cookie.

Explicit declare headers need to operate.

Reply:
updated